src/plan_partialize.c

```diff
@@ -167,19 +167,20 @@ partialize_agg_paths(RelOptInfo *rel)
  * simple filter (e.g., HAVING device > 3). In such cases, the HAVING clause is
  * removed and replaced by a filter on the input.
  */
-void
+bool
 ts_plan_process_partialize_agg(PlannerInfo *root, RelOptInfo *input_rel, RelOptInfo *output_rel)
 {
 	Query *parse = root->parse;
-
+	bool partials_found = false;
 #if PG10_GE
 	Assert(IS_UPPER_REL(output_rel));
 #endif
 
 	if (CMD_SELECT != parse->commandType || !parse->hasAggs)
-		return;
+		return partials_found;
 
-	if (has_partialize_function(parse))
+	partials_found = has_partialize_function(parse);
+	if (partials_found)
```

Comment:
Nit: would suggest to refactor this to the form:

```c
if (!has_partialize_function(parse))
  return false;
<old code>
return true
```

This is easier to follow and less error-prone.

Reply:
:+1: